### PR TITLE
Change with-violation to violation for consistency

### PR DESCRIPTION
**Pull request by yiraeChristineKim**
Change `with-violation` to `violations`.
Change `without-violation` to `No violations`
Remove the violation row in policytemplate detail
Change `NonCompliant` to `violations`
Change `Compliant` to `No violations`
Signed-off-by: yiraeChristineKim <yikim@redhat.com>

### CODE REVIEW

**Review by JustinKuli — APPROVED** [review]

**Review by dhaiducek — COMMENTED** [review]
Is there an issue to attach here?

Also, having "3 No violations" seems odd, but I'm definitely on board with consistency! 😄

File: frontend/src/routes/Home/Overview/OverviewPage.test.tsx [hunk 1]
```diff
@@ -1115,8 +1115,8 @@ it('should render overview page with expected data', async () => {
 
   // Check Cluster violations chart rendered
   await waitFor(() => expect(getAllByText('Cluster violations')).toHaveLength(2))
-  await waitFor(() => expect(getByText('1 Without violations')).toBeTruthy())
-  await waitFor(() => expect(getByText('1 With violations')).toBeTruthy())
+  await waitFor(() => expect(getByText('1 No violations')).toBeTruthy())
```
Comment (mprahl) [inline frontend/src/routes/Home/Overview/OverviewPage.test.tsx:1118]: Can we change this particular case to be `with no violations` whenever there is a number before `No violations`?

**Comment by dhaiducek** [comment]
Oops. Forgot about that bug in Tide that doesn't wait for the additional reviewer that was required 😅

**Comment by JustinKuli** [comment]
My bad :disappointed:

**Pull request merged**